docker/Dockerfile.amd

```diff
@@ -35,6 +35,7 @@ RUN rm /etc/apt/sources.list.d/* \
     gfortran \
     flex \
     hwloc \
+    gdb \
```

Comment:
Why not in the base dockerfile?

and, random thought, could we also add 

* python3-dbg (for gdb python support)
* valgrind

they should be pretty small, but very helpful.

Thoughts?